test/FastTests/Voron/CompactTrees/RavenDB-21521.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using FastTests.Voron.FixedSize;
+using Raven.Server.Documents.Indexes.Static;
+using Sparrow;
+using Tests.Infrastructure;
+using Voron.Data.CompactTrees;
+using Voron.Global;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace FastTests.Voron.CompactTrees
+{
+    public class RavenDB_21521 : StorageTest
+    {
+        public RavenDB_21521(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        public class RandomTrainIterator : IReadOnlySpanEnumerator
+        {
+            private ReadOnlySpan<byte> Chars => "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"u8;
+            private Random _generator;
+            private int _currentIdx;
+            private int _maxSize;
+
+
+            public RandomTrainIterator(int count, int maxSize, int seed)
+            {
+                _generator = new Random(seed);
+
+                _currentIdx = 0;
+                _maxSize = maxSize;
+
+                Count = count;
+            }
+
+            public int Count { get; }
+
+            public void Reset()
+            {
+                _currentIdx = 0;
+            }
+
+            public bool MoveNext(out ReadOnlySpan<byte> result)
+            {
+                if (_currentIdx > Count)
+                {
+                    result = ReadOnlySpan<byte>.Empty;
+                    return false;
+                }
+
+                var output = new byte[_generator.Next(_maxSize)];
+
+                for (int i = 0; i < output.Length; i++)
+                    output[i] = Chars[_generator.Next(Chars.Length)];
+
+                _currentIdx++;
+
+                result = output.AsSpan();
+                return true;
+            }
+        }
+
+        [RavenTheory(RavenTestCategory.Voron)]
+        [InlineDataWithRandomSeed]
+        public void CompactKeyFuzzy(int seed)
```

Comment:
Has this happened with every seed? Maybe it's worth adding seeds that have reproduced the problem?

Reply:
I agree, we also need to have test cases with the seeds for which it failed

Reply:
All of them reproduce it :D .. after figuring out how to trigger it was the most consistent fuzzy test I have seen.